app/models/shelter.rb

```diff
@@ -31,4 +32,15 @@ def alphabetical_pets
   def shelter_pets_filtered_by_age(age_filter)
     adoptable_pets.where("age >= ?", age_filter)
   end
+
+  def self.reverse_abc
+    find_by_sql("
+      SELECT * FROM shelters
+      ORDER BY name DESC
+    ")
+  end
+
+  def self.pending_applications
+    Shelter.joins(pets: :applications).where("applications.status = ?", "Pending").order(:name)
```

Comment:
since you have this shortcut, you can actually do `Shelter.joins(:applications)`